### PR TITLE
fix: patch-package is not applied in dist'ed build

### LINKED ISSUE #19083: Cypress 9.1 open wrong URL when there is a ~ (tilde) in the link
### Current behavior

In Cypress 9.1, some of our tests failed due to Cypress process the URL in `visit()` calls incorrectly.

This code:

```
cy.visit('/?q=~%28match_status~%28~%27invoiced%29%29')
```

causes Cypress to visit this link:

```
/?q=%5Cx7e%28match_status~%28~%27invoiced%29%29
```

Note the strange `%5Cx7e` right after the `q=`.

### Desired behavior

Cypress should not alter the visit URL in anyway.

### Test code to reproduce

https://github.com/lephuongbg/cypress-test-tiny/tree/cypress-9.1-url-bug

Quick link to diff:

https://github.com/cypress-io/cypress-test-tiny/compare/master...lephuongbg:cypress-9.1-url-bug

### Cypress Version

9.1.0

### Other

_No response_

### LINKED ISSUE #19084: Cypress 9.1 appears to mishandle POST content with special characters
### Current behavior

Upgraded from 9.0 to 9.1

Multiple Tests that send a POST including random generated password content with blackslashes:

e.g.  "oldPassword":"S.]\\x7e}gZ3=kFJ"  (Note the double backslash - GH editor might hide this!).

Fails at when received at the server as oldPassword is wrong. 

### Desired behavior

Tests execute POST commands as before.

### Test code to reproduce
-UPDATED-
Analysis as in comment below from slavab89 appears to be what I am observing.

### Cypress Version

9.1.0

### Other

Thank you. 😄

### LINKED ISSUE #19087: Cypress 9.1 seems to not be reporting to dashboard
### Current behavior

I'm not sure if I'm the only one with the issue, but 9.0 works fine, but 9.1 does not report to dashboard (batch build@codebuild if that matters) and throws

```
The run started, but never completed. This can happen when the run is canceled from CI or when Cypress crashes during running tests.

Check your Awscodebuild logs for more information. Archive this run to remove it from the runs list and analytics.
```

### Desired behavior

_No response_

### Test code to reproduce

Run batch build  @ 9.1 on Codebuild

### Cypress Version

9.1

### Other

_No response_

### LINKED ISSUE #19091: Possible regression in Cypress 9.1: `TypeError: data.hasOwnProperty is not a function`
### Current behavior

We're running a series of tests against our production application using Cypress. The runner is using the `cypress/browsers:node16.5.0-chrome94-ff93` Docker image. All of our tests fail with a similar stacktrace and none of the lines include our own test code. This suite runs successfully in Cypress 9.0.

Here's the failure we get on almost all of the specs in our suite:

```
TypeError: data.hasOwnProperty is not a function
--
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:39:22)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:32:26)
  | at Object.deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:16:17)
  | at Encoder.encodeAsBinary (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/index.js:80:41)
  | at Encoder.encode (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/index.js:42:29)
  | at Client._packet (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/client.js:169:44)
  | at Socket.packet (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:160:21)
  | at /root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:274:18
  | at /root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/lib/socket-base.js:318:28
  | at tryCatcher (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
  | at Promise._settlePromiseFromHandler (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:547:31)
  | at Promise._settlePromise (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
  | at Promise._settlePromise0 (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
  | at Promise._settlePromises (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:725:18)
  | at _drainQueueStep (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:93:12)
  | at _drainQueue (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:86:9)
  | at Async._drainQueues (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:102:5)
  | at Immediate.Async.drainQueues [as _onImmediate] (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:15:14)
  | at processImmediate (node:internal/timers:464:21)
  |  
  | TypeError: data.hasOwnProperty is not a function
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:39:22)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:40:32)
  | at _deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:32:26)
  | at Object.deconstructPacket (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:16:17)
  | at Encoder.encodeAsBinary (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/index.js:80:41)
  | at Encoder.encode (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/index.js:42:29)
  | at Client._packet (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/client.js:169:44)
  | at Socket.packet (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:160:21)
  | at /root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:274:18
  | at /root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/lib/socket-base.js:318:28
  | at tryCatcher (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
  | at Promise._settlePromiseFromHandler (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:547:31)
  | at Promise._settlePromise (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
  | at Promise._settlePromise0 (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
  | at Promise._settlePromises (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:725:18)
  | at _drainQueueStep (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:93:12)
  | at _drainQueue (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:86:9)
  | at Async._drainQueues (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:102:5)
  | at Immediate.Async.drainQueues [as _onImmediate] (/root/.cache/Cypress/9.1.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:15:14)
  | at processImmediate (node:internal/timers:464:21)
  |  
  | [502:1124/150621.083113:ERROR:bus.cc(392)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
  | [502:1124/150621.084970:ERROR:bus.cc(392)] Failed to connect to the bus: Address does not contain a colon
  | [502:1124/150621.085005:ERROR:bus.cc(392)] Failed to connect to the bus: Address does not contain a colon
  | [666:1124/150621.113215:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is
```

### Desired behavior

We're just looking to figure out what changed from 9.0 -> 9.1 to cause this issue. We're able to run these tests locally, but they fail in 

### Test code to reproduce

```
  it('Home page - shows expected content', () => {
    cy.visit('/');

    cy.contains('Making millions of students better readers and writers');
  });
```

Our base URL is www.commonlit.org.

### Cypress Version

9.1.0

### Other

_No response_

### LINKED ISSUE #19096: setCookies is broken since 9.0.0. `~` is converted to `\x7e`
### Current behavior

After calling setCookie with a valid cookie and then loading the desired page the cookies are somehow not accepted or read. The page we load should show the user is logged in, but he appears as if the cookies are not there. The Network Application Cookies show that they are set, but the page that is loaded doesn't seem to either see them or read them. Note in this example baseUrl is not set, but I tried it with it being set and not. Same behavior.

I ran 8.7.0 and 9.0.0 and 9.0.1. Its broken since 9.0.0. Tested on Chrome, Firefox and Electron. Broken in all 3. 
In 8.7.0 the user appears to be logged in, in 9.* the user is not.

### Desired behavior

User should be logged in after calling setCookies and then visiting the destination url. In this example when I run the provided code and www.yahoo.com is loaded and I move the mouse over the top right `mail` icon I see the user as logged in.  

See screen shot for Cypress 8.7.0
Screen Shot 2021-11-24 at 11 34 07 AM

Now when running with 9.0.0 you will see this. As if the user is not signed in.
Screen Shot 2021-11-24 at 11 31 42 AM

### Test code to reproduce

I can't give you fully reproducible code, because I can't provide you with the way we get our cookies. Here is the basic script with a hardcoded sample cookie value. 

```
describe('Main Dashoard login load test', function() {
  context('Login', function() {
    it('When moving the mouse over the mail icon at the top right corner the user should show logged in', function() {
      cy.clearCookie('Y');
      cy.clearCookie('T');
      cy.getCookie('Y').should('not.exist');
      cy.getCookie('T').should('not.exist');

      cy.setCookie('Y', "v=1&np=1&n=23eno0bmig3ub&l=8a2dn1nck044hib2atdgsc5adwppb5ii12lf2i04/o&p=n33vvvv00000000&r=11p&intl=us");
      cy.setCookie('T', "af=JnRzPTE2Mzc3Nzg4NTYmcHM9SDR2cHdfbnZXdd=bnMBeWFob28BZwFDTU5NWlhOUURBNkI3SFoyTFlTTUpLQUY3RQFhYwFBSTJLRmlHNQFhbAFkc3BfYXV0bwFzYwF5cWEBZnMBa1VGSTV3Qmhub1dvAXp6AW9Xb25oQjhBQQFhAVFBRQFsYXQBb1dvbmhCAW51ATA-&kt=asdffadfaf--~I&ku=FAATcU6Cc1YDmlpoXPv7S4BTWHg5Uz58mDuWHcxOtOQKf6gvdQ10fhJAnSJ9k_HlToJ4.YDa4kNdee2LAEpqg.KtqhyHvvi2o5MRjmhOesmE7p4fxxHAwiK1gYjNXygVm_AHV95GryLqDPnfuEFaHFQy_vfOKU1YtxEi3W2AdQM3j8-~E");
      cy.visit('https://www.yahoo.com');
    });
  });
});
```

### Cypress Version

9.*

### Other

_No response_

### LINKED ISSUE #19105: Cypress 9.1.0 freezes if baseUrl changes between tests
### Current behavior

In my tests, it is necessary to open an html file locally in one of the tests. It is impossible to do this with baseUrl, so for this test I reset baseUrl, but subsequent tests continue to use baseUrl, which is set globally in cypress.json. I saw such a solution here - https://github.com/cypress-io/cypress/issues/4450#issuecomment-778201514
In cypress 9.0.0 and below, this method worked, but in 9.1.0, the execution of tests freezes if, after a test with base Url = null, a test is run in which baseUrl is set globally. At the same time, the test runner continues to work until it is stopped manually. There is an error in the console
image

[www.cypress.io-1637826397522.log](https://github.com/cypress-io/cypress/files/7601100/www.cypress.io-1637826397522.log)

### Desired behavior

The execution of the tests continues correctly, just as it was in version 9.0.0.
And it is better to make it possible to visit local files with an existing baseUrl

### Test code to reproduce

You can use any local html file for the test
```
it('Base url = null', { baseUrl: null }, () => {
  cy.visit('cypress/downloads/test.html')
})

it('Base url = https://www.cypress.io/', () => {
  cy.visit('/')
})

```
in cypress.json
```
"baseUrl": "https://www.cypress.io/"
```

### Cypress Version

9.1.0

### Other

_No response_

### LINKED ISSUE #19115: Bad body encoding in `cy.request`
### Current behavior

Body is encoding chars like  '~' into hex chars (` \x7e`)

### Desired behavior

Body sent should be ` { key: '~'}` without encoding

### Test code to reproduce

```js
    cy.request({ method: 'POST', body: { key: '~'}, 'url': 'https://www.google.com'})

```

### Cypress Version

9.1.0

### Other

I thank you for what you do at Cypress ! I'm facing the issue described here since the v9.1, is it a bug or is it what you wanted? 😬 

Thanks for your time !

### LINKED ISSUE #19178: cy.intercept fails to respond with JSON containing tilde (~) in string property
### Current behavior

When intercepting a request and responding with a JSON that contains a tilde an error is thrown.
```
cy.intercept("GET", "localhost/123", (res) => {
      res.reply({
        bla: "~",
      });
    });
```

`VM263:1 Uncaught (in promise) SyntaxError: Unexpected token x in JSON at position 9`

This seems to be a regression in 9.1.0, in 9.0.0 this executes just fine.

### Desired behavior

No error, the response contains the correct object with a tilde.

### Test code to reproduce

Repro repo: https://github.com/LoaderB0T/cypress-tilde-regression

### Cypress Version

9.1.0

### Other

_No response_

### PR DESCRIPTION
**Pull request by tgriesser**
- Closes #19091
- Closes #19083 
- Closes #19084
- Closes #19115
- Closes #19096
- Closes #19178
- Closes #19087
- Closes #19105

### User facing changelog

Due to changes to build process changes in #17285, the `patch-package` scripts were not being properly applied as they were previously. Fixes the build process to include the `scripts` entries in both the root & packages for the yarn install, so the postinstall entries are applied properly.

Added tests in `testStaticAssets.ts` to ensure the correct patches are being applied.

### CODE REVIEW

File: packages/socket/package.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
   },
   "dependencies": {
     "circular-json": "0.5.9",
+    "engine.io": "5.0.0",
+    "engine.io-parser": "4.0.2",
```
Comment (tgriesser) [inline packages/socket/package.json:18]: Needed to add this, as it was being hoisted when devDeps are stripped & patch wasn't being applied.

File: packages/server/package.json [hunk 2]
```diff
@@ -113,6 +113,7 @@
     "trash": "5.2.0",
     "tree-kill": "1.2.2",
     "ts-node": "8.5.4",
+    "tsconfig-paths": "3.10.1",
```
Comment (emilyrohrbough) [inline packages/server/package.json:116]: why did this need move from a dev-dep?

File: scripts/binary/build.ts [hunk 3]
```diff
@@ -68,15 +68,23 @@ export async function buildCypressApp (options: BuildCypressAppOpts) {
   // Copy Packages: We want to copy the package.json, files, and output
```
Comment (emilyrohrbough) [inline scripts/binary/build.ts:68]: why re-name from `.patch` to `.dev.patch`?

Comment (emilyrohrbough) [reply scripts/binary/build.ts:68]: Do the patches in `packages/socket/patches` & `packages/serverp/patches` also need to be renamed to `.dev.patch`?

Comment (tgriesser) [reply scripts/binary/build.ts:68]: https://github.com/cypress-io/cypress/pull/19239/files#diff-d4d26f3d914f934bada7003fc6d0be8ebc4a24777a765c0302a1ece4892c2cf2R85

It's the way you're supposed to name them for `devDependencies`

Comment (emilyrohrbough) [reply scripts/binary/build.ts:68]: Ahhhh okay 👍🏻

Comment (tgriesser) [reply scripts/binary/build.ts:68]: Oops, had the wrong link. But here's the docs: https://github.com/ds300/patch-package#dev-only-patches

And no, the socket & server ones are prod dependencies, so they remain `.patch`

**Review by chrisbreiding — DISMISSED** [review]

**Review by emilyrohrbough — DISMISSED** [review]

File: packages/server/package.json (hunk 2 above)
Comment (tgriesser) [reply packages/server/package.json:116]: Trying to find where it failed - might have only been locally when I was writing the test case for the patch being applied, and it wasn't... and then I noticed that we do indeed use the package in a prod situation: https://github.com/cypress-io/cypress/blob/11e99fc35a4cf1fe43abed07cd3f7fb2441fa461/packages/server/lib/util/ts_node.js#L30-L37

Comment (emilyrohrbough) [reply packages/server/package.json:116]: Nice catch then!

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**